pwndbg/aglib/vmmap_custom.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import bisect
+from typing import List
+from typing import Set
+from typing import Tuple
+
+import pwndbg
+import pwndbg.aglib.memory
+import pwndbg.aglib.stack
+import pwndbg.color.message as M
+import pwndbg.lib.cache
+import pwndbg.lib.config
+import pwndbg.lib.memory
+from pwndbg.dbg import EventType
+
+# List of manually-explored pages which were discovered
+# by analyzing the stack or register context.
+explored_pages: List[pwndbg.lib.memory.Page] = []
+
+# List of custom pages that can be managed manually by vmmap_* commands family
+custom_pages: List[pwndbg.lib.memory.Page] = []
+
+auto_explore = pwndbg.config.add_param(
+    "auto-explore-pages",
+    "warn",
+    "whether to try to infer page permissions when memory maps missing (can cause errors)",
+    param_class=pwndbg.lib.config.PARAM_ENUM,
+    enum_sequence=["yes", "warn", "no"],
+)
+
+
+_warn_cache: Set[int] = set()
+
+
+@pwndbg.dbg.event_handler(EventType.NEW_MODULE)
+def clear_warn_cache():
+    _warn_cache.clear()
+
+
+def get_custom_pages() -> Tuple[pwndbg.lib.memory.Page, ...]:
+    """
+    Returns a tuple of `Page` objects representing the memory mappings of the
+    target, sorted by virtual address ascending.
+    """
+    pages: List[pwndbg.lib.memory.Page] = []
+    pages.extend(explored_pages)
+    pages.extend(custom_pages)
+    pages.sort()
```

Comment:
We can do:
```py
pages: List[pwndbg.lib.memory.Page] = [*explored_pages, *custom_pages]
pages.sort()
return tuple(pages)
```

Reply:
Could also be: `tuple(sorted(explored_pages+custom_pages))` hmm...

Reply:
done